packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -1756,6 +1759,13 @@ class FocusManager with DiagnosticableTreeMixin, ChangeNotifier {
   void _applyFocusChange() {
     _haveScheduledUpdate = false;
     final FocusNode? previousFocus = _primaryFocus;
+    for (final _Autofocus autofocus in _pendingAutofocuses) {
+      if (identical(autofocus.scope._manager, this)) {
```

Comment:
@LongCatIsLooong did you mean this line? For the test I posted this is true .

Reply:
Ohh right. I thought detaching removes the manager.

Reply:
Maybe we can do this.

Reply:
After looking at `FocusNode.detach`, I think the idea is the widget tree (or whatever that the focus tree is going to attach to & and manage focus for) is responsible for calling "reparent" and "detach" so each focus node has the right `parent` and `manager`. So I think these checks should be sufficient.

@xu-baolin you mean this is true for something in the first row? I'd expect those elements to have already been disposed of (along with their focus nodes)?

Reply:
The node is disposed but the scope node not. 
```dart
if (identical(autofocus.scope._manager, this) && autofocus.autofocusNode != null)
```